src/plugins/xerces/deserializer.cpp

```diff
@@ -0,0 +1,141 @@
+/**
+ * @file
+ *
+ * @brief deseialization implementation for xerces plugin
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ */
+
+#include "deserializer.hpp"
+
+#include <fstream>
+#include <functional>
+#include <iostream>
+#include <memory>
+
+#include <xercesc/dom/DOM.hpp>
+#include <xercesc/dom/DOMAttr.hpp>
+#include <xercesc/dom/DOMDocument.hpp>
+#include <xercesc/dom/DOMError.hpp>
+#include <xercesc/dom/DOMException.hpp>
+#include <xercesc/dom/DOMImplementation.hpp>
+#include <xercesc/dom/DOMNamedNodeMap.hpp>
+#include <xercesc/dom/DOMNode.hpp>
+#include <xercesc/parsers/XercesDOMParser.hpp>
+
+#include <key.hpp>
+
+XERCES_CPP_NAMESPACE_USE
+using namespace std;
+using namespace kdb;
+
+/*
+ + Smart pointer helper functions to ease the pain of dealing between XMLCh and std::string
+ */
+
+// no need to expand this, just keep it in one line
+// TODO move to external file if also used for serializer
+// clang-format off
+struct XmlChDeleter { void operator() (XMLCh * ptr) { XMLString::release (&ptr); } };
+
+struct StringDeleter {
+	// Workaround as ptr->c_str gives as a const * which we cant free
+	char * cStr;
+	// TODO test if this really frees the str too, as it seems to do
+	// delete str afterwards says its already freed
+	void operator() (string * ptr) { XMLString::release (&cStr); }
+};
+
+// clang-format on
+
+static unique_ptr<XMLCh, XmlChDeleter> fromStr (const std::string str)
+{
+	return unique_ptr<XMLCh, XmlChDeleter> (XMLString::transcode (str.c_str ()));
+}
+
+static unique_ptr<std::string, StringDeleter> toStr (XMLCh const * xmlCh)
+{
+	char * cStr = XMLString::transcode (xmlCh);
+	return unique_ptr<std::string, StringDeleter> (new string (cStr), StringDeleter{ cStr });
+}
+
+/*
+ * Actual Xerces logic
+ */
+
+static unique_ptr<XercesDOMParser> doc2dom (const std::string src)
+{
+	unique_ptr<XercesDOMParser> parser (new XercesDOMParser);
```

Comment:
Thats not possible as the copy constructor of XercesDOMParser is private, looks like its not intended to be done that way.